website/src/services/subscribeService.ts

```diff
@@ -0,0 +1,32 @@
+// src/services/subscribeService.ts
+
+import { validateEmail } from '@/utils/validateEmail'; // Import the validation utility
+
+/**
+ * Handles the logic for subscribing an email, including validation.
+ * @param email - The email address to subscribe.
+ */
+export async function handleSubscription(email: string): Promise<void> {
```

Comment:
**question:** Why not using a [server action](https://nextjs.org/docs/app/building-your-application/data-fetching/server-actions-and-mutations) here ? This would be make the code way easier to maintain ver time instead of splitting service / api